iwyu_preprocessor.cc

```diff
@@ -751,6 +758,23 @@ void IwyuPreprocessorInfo::FileChanged_ExitToFile(SourceLocation include_loc,
          "begin_exports without an end_exports");
     begin_exports_location_stack_.pop();
   }
+
+  const FileEntry* includer = GetFileEntry(associated_pragma_location_);
```

Comment:
My reasoning for exit to file over FileChanged_EnterFile was on my assumption that HandlePragmaComment was called after FileChanged_EnterFile, and not before.
After debugging this seems to not be the case, so i will try to move this code to the AddDirectInclude() function, which also has the other associated header handling.